agent/agent_test.go

```diff
@@ -95,4 +95,85 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, tc, cleanup := agentTestEnv(t)
+	defer tc.Stop()
+	defer cleanup()
+
+	sessionMessage := api.SessionMessage{SessionID: "sm1",
+		Node: &api.Node{},
+		Managers: []*api.WeightedPeer{
+			{
+				Peer:   &api.Peer{NodeID: "node1", Addr: "10.0.0.1"},
+				Weight: 1.0}},
+		NetworkBootstrapKeys: []*api.EncryptionKey{{}}}
+	err := agent.handleSessionMessage(ctx, &sessionMessage)
```

Comment:
@rajdeepd Not at all. In fact, to properly test this, we'd have to inject a dispatcher client, so it would take some refactoring.

Let's do this:

1. Turn this into a table based test. It is just testing that there aren't errors for common structures, so it will be excellent to have these laid out.
2. Place a TODO here that testing this internal interface is non-ideal and means we should refactor agent session message handling to be more testable.

The main issue here is that because of the way this test is structured, its clear the design isn't quite there. If these two things are done, we can carry on and address these problems without having to worry too much about unit test complexity.